server.js

```diff
@@ -223,10 +224,9 @@ if (process.env.DEBUG_DUMMY_SMTP) {
   }
 }
 
-if (gTransporter) {
-  app.listen(port, function() {
-    console.log("Listening on " + port);
-  });
-} else {
-  console.error("SMTP credentials couldn't be read from the environment, exiting.");
+if (!gTransporter) {
+  console.log("SMTP config unavailble. Email features will not work.")
```

Comment:
typo: "unavailable"

Reply:
Also, what happens when we set `gTransporter` to null above on L221?
I think it'll blow up above on L123 where we try and do `gTransporter.sendMail(...)` and we'll need to maybe set the `gTransporter` object to a dummy object like on L197-201.

Reply:
I'd prefer simply failing and exiting in this case for now.

Reply:
I ran into this when I tried to deploy the server to a new heroku app to test Firefox Accounts. Requiring a fully-configured SMTP backing service means it's impossible to work on any other part of the app - e.g., Firefox Accounts, front-end web pages, REST API, scheduled jobs/processes - unless you first configure an SMTP backing service.

Reply:
But that's the purpose of the dummy SMTP mode flag. I added that for a "clean" way to explicitly opt-in to a "dev mode" of sorts that doesn't require an SMTP service. Let me know what you think, or if I'm missing something.